backend/base/logging.py

```
@@ -100,10 +101,18 @@ def shouldRollover(self, record: logging.LogRecord) -> int:
 }
 
 
-def setup_logging(do_rollover: bool = True) -> None:
+def setup_logging(
+    log_folder: str = None,
+    do_rollover: bool = True,
+) -> None:
     "Setup the basic config of the logging module"
 
-    LOGGING_CONFIG["handlers"]["file"]["filename"] = get_log_filepath()
+    if log_folder is None:
+        from backend.base.files import folder_path
+        LOGGING_CONFIG["handlers"]["file"]["filename"] = folder_path(Constants.LOGGER_FILENAME)
+    else:
+        LOGGING_CONFIG["handlers"]["file"]["filename"] = join(log_folder, Constants.LOGGER_FILENAME)
```

No validation is done whether the log folder exists, and whether it's a folder (and not a file). If that's not the case, it should raise a Value error that's caught in Kapowarr.py and processed at the same place where the database folder error is caught.

---

Mate please read my entire message. You forgot half of it:

> [...] that's caught in Kapowarr.py and processed at the same place where the database folder error is caught.